### ISSUE TITLE
std::random_shuffle deleted in C++17 under MSVC.

### ISSUE THREAD

**Issue by dkeeney** [htm-community/nupic.cpp#223]
When I compile bindings\math\py_SparseBinaryMatrix.cpp on MSVC I get the compile error:
```
  c:\projects\ai\cppw\src\nupic\math\sparsebinarymatrix.hpp(126): error C2039: 'random_shuffle': is not a member of 'std' (compiling source file C:\Projects\AI\cppW\bindings\py\cpp_src\bindings\math\py_SparseBinaryMatrix.cpp) 
```
https://stackoverflow.com/questions/45013977/random-shuffle-is-not-a-member-of-std-error
suggests that we should use the following:
```
std::random_device rd;
std::mt19937 g(rd());

std::shuffle(v.begin(), v.end(), g);
```

@breznak I know you worked on the random number generator.  Can I call nupic::Random::shuffle(first, last) instead? See:
- nupic/math/SparseBinaryMatrix.hpp line 126 
- nupic/math/sparsematrix.hpp line 1326

**Comment by breznak** [comment]
> an I call nupic::Random::shuffle(first, last) instead? See: 

@dkeeney yes, exactly. Random::shuffle is prefered (even required), as std:: is not platform independent. 


>   // our reimpementation of std::shuffle,
  // as the implementation is not dictated by the standard,
  // resuting in differences between impementations (OS, stdlib,...) :(
  // https://en.cppreference.com/w/cpp/algorithm/random_shuffle

**Comment by breznak** [comment]
this seems to have been fixed in #222
